cli/app/commands/conflict/conflict.py

```diff
@@ -0,0 +1,391 @@
+        try:
+            # Handle simple version comparisons (backwards compatibility)
+            if not any(op in expected for op in [">=", "<=", ">", "<", "==", "!=", "~", "^"]):
+                # Default to >= for simple version strings
+                return version.parse(current) >= version.parse(expected)
+
+            # Handle version ranges and specifiers
+            spec_set = SpecifierSet(expected)
+            return Version(current) in spec_set
+
+        except Exception:
+            # Fallback to string comparison
+            return current == expected
+
+    @staticmethod
+    def normalize_version_requirement(requirement: str) -> str:
+        """Parse version requirement and return a normalized specifier."""
+        if not requirement:
+            return requirement
+
+        # Handle npm-style caret ranges: ^1.20.0 -> >=1.20.0, <2.0.0
+        if requirement.startswith("^"):
+            base_version = requirement[1:]
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                return f">={base_version}, <{major + 1}.0.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Handle npm-style tilde ranges: ~1.20.0 -> >=1.20.0, <1.21.0
+        if requirement.startswith("~"):
+            base_version = requirement[1:]
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                minor = parsed.minor
+                return f">={base_version}, <{major}.{minor + 1}.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Handle ranges with commas: 1.20.0,2.0.0 -> >=1.20.0, <2.0.0
+        if "," in requirement and not any(op in requirement for op in [">=", "<=", ">", "<", "==", "!="]):
+            parts = [p.strip() for p in requirement.split(",")]
+            if len(parts) == 2:
+                return f">={parts[0]}, <{parts[1]}"
+
+        # Handle x.x.x format ranges: 1.20.* -> >=1.20.0, <1.21.0
+        if requirement.endswith(".*") or requirement.endswith(".x"):
+            base_version = requirement.replace(".*", ".0").replace(".x", ".0")
+            try:
+                parsed = version.parse(base_version)
+                major = parsed.major
+                minor = parsed.minor
+                return f">={base_version}, <{major}.{minor + 1}.0"
+            except Exception:
+                return f">={base_version}"
+
+        # Return as-is if it already contains operators
+        return requirement
+
+
+class ConfigLoader:
+    """Handles loading and parsing of configuration files."""
+    
+    def __init__(self, logger: LoggerProtocol):
+        self.logger = logger
+
+    def load_config(self, config_path: str) -> Dict[str, Any]:
+        """Load configuration from YAML file."""
+        self.logger.debug(conflict_loading_config.format(path=config_path))
+
+        if not os.path.exists(config_path):
```

Comment:
make use of the `lib.py` for these kind of utility function, there are already code similar to this,